best-practices.md

```diff
@@ -32,6 +33,19 @@ the value consist of only lowercase characters, numbers, `_`, and `-`. Examples
 `landsat-8` (Landsat-8) and `envisat` (Envisat). This is to provide consistency for search across collections, so that
 people can just search for 'landsat-8', instead of thinking through all the ways providers might have chosen to name it.
 
+## Field selection and Metadata Linking
+
+In general STAC aims to be oriented around **search**, centered on the core fields that users will want to search on to find 
+imagery. The core is space and time, but there are often other metadata attributes that are useful. While the specification is 
+flexible enough that providers can fill it with tens or even hundreds of fields of metadata, that is not recommended. If 
+providers have lots of metadata then that can be linked to in the [Asset Object](item-spec/item-spec.md#asset-object) 
+(recommended) or in a [Link Object](item-spec/item-spec.md#link-object). There is a lot of metadata that is only of relevance 
+to loading and processing data, and while STAC does not prohibit providers from putting those type of fields in their items, 
+it is not recommended. For very large 
+catalogs (hundreds of millions of records), every additional field that is indexed will cost substantial money, so data
```

Comment:
Although I basically agree with the indexing issue, I doubt that people blindly index everything, do they? Like you could skip everything complex (i.e. objects like bands) because the API can't search them anyway. If I'd implement something like that, I'd probably implement a white list of fields to be indexed beforehand...

Reply:
People definitely blindly index everything. Oftentimes the person doing the index is a dev who has no idea about geospatial or the use case of search. And they're asked by someone who has no idea about software/databases/indexes, and when the dev asks them if they should index everything they'd be inclined to say yes, as they don't understand the effects, and why not? That's why I think it's important to have the recommendation - to spark the conversation between diverse people.

Reply:
I know which fields should be indexed, and I still index everything, just because I hadn't gotten around to figuring out how I should be indexing certain fields in Elasticsearch but still store the entire record.